client/src/routes/PageSearch.js

```diff
@@ -18,6 +18,36 @@ import SearchEasterEggs from '../utilities/SearchEasterEggs';
 
 
 class PageSearch extends React.Component {
+  async componentDidMount() {
+    const p = this.props;
+
+    if(window.location.search.length !== 0) {
+      const urlParams = new URLSearchParams(window.location.search);
+
+      if (!p.search.filterOptions) {
+        await p.dispatch(fetchFilterOptions())
```

Comment:
Ah, now that I see the code better--you probably should use the `componentDidUpdate` strategy I alluded to instead, since I'd like to avoid double fetching with the code in `SearchForm` (which could happen in a race, since if SearchForm loads before this fetch completes, then both will call the action).

You can use the `isFetching` prop you mentioned and keep checking for it (and you won't need async/await anymore).

Reply:
Also, as a gotcha, note that componentDidUpdate does not include the initial mount, but that should be fine, since you only want to fire after the props change (isFetching goes from true -> false) anyways.

Reply:
Wouldn't waiting for `isFetching` to change also cause a race? In the event that it has already changed from `true` to `false` before the component is loaded.